### PR TITLE
Fixes Event Log Message Format on Unmanaged FailFast

### PR DESCRIPTION
**Pull request by Silvenga**
We observed the following event log messages: 

```
CoreCLR Version: 6.0.21.52210
.NET Version: 6.0.0
Description: The process was terminated due to an internal error in the .NET Runtime.at IP 01090791 (00F40000) with exit code 80131506.
```

We are still investigating the crashes, but in the meantime we noticed the message was a little off, this PR just fixes the fallback messages to match the newer versions in resources (assumed) to make the message read a bit nicer. 😸

### CODE REVIEW

File: src/coreclr/vm/eventreporter.cpp [hunk 1]
```diff
@@ -110,7 +110,7 @@ EventReporter::EventReporter(EventReporterType type)
 
     case ERT_ManagedFailFast:
         if(!ssMessage.LoadResource(CCompRC::Optional, IDS_ER_MANAGEDFAILFAST))
-            m_Description.Append(W("Description: The application requested process termination through Environment.FailFast."));
+            m_Description.Append(W("Description: The application requested process termination through System.Environment.FailFast(string message)."));
```
Comment (jkotas) [inline src/coreclr/vm/eventreporter.cpp:113]: `Environment.FailFast` has multiple overloads. I do not think that this change is an improvement.

Comment (Silvenga) [reply src/coreclr/vm/eventreporter.cpp:113]: Is the resource version wrong then?

Comment (jkotas) [reply src/coreclr/vm/eventreporter.cpp:113]: Ah ok. I see that you have just copied over the string from resources.

Comment (jkotas) [reply src/coreclr/vm/eventreporter.cpp:113]: I would delete "(string message)" from both here and the resource.

Comment (Silvenga) [reply src/coreclr/vm/eventreporter.cpp:113]: Yeah, we assumed it was just forgotten during another reflector. 

Will do!

**Comment by danmoseley** [comment]
Being picky, it seems to me that `IP 01090791 (00F40000) with exit code 80131506.` ought to be `IP 0x01090791 (0x00F40000) with exit code 0x80131506.` so it's clear they're all hex.

This formats with %p, %p and %x respectively. A gotcha is %p prefixes `0x` when I try with clang, and not when I try with cl. %x is never prefixed.

**Comment by danmoseley** [comment]
Test failure fix is in https://github.com/dotnet/runtime/pull/63529

**Comment by Silvenga** [comment]
@danmoseley - I updated the formats with `0x%x` to match other places that hex is being formatted, that seems to be the most portable approach.

**Review by jkotas — APPROVED** [review]
Thank you!

**Comment by Silvenga** [comment]
Unrelated, the crashes were caused by us 😭 - slight incompatibility with changes in the span implementation in .NET 6 😆.

**Pull request merged**